### ISSUE TITLE
Krunner: KNS installation category

### ISSUE THREAD

**Issue by star-buck** [blue-systems/plasma-ideas#55]
Added new category as someone suggested here and sorted some Runners to it:
https://www.reddit.com/r/kde/comments/63nlkg/kde_store_should_create_an_extra_category_for/

We could now think if we like to add a GHNS button:
https://store.kde.org/browse/cat/375/ord/latest/

**Comment by notmart** [comment]
problem is, krunner plugins are binary plugins written in c++, so can't be installed by knewstuff .
from what i see in the store category, there are 3 runner written for plasma5 (the top 3 entries) all the rest seems to be kde4 stuff.
would be needed some kind of bindings to write scripted-only runners, then provide only those in the store

**Comment by aleixpol** [comment]
Also we could allow plugins through dbus, then people would be able to write their plugins in whichever technology they please.

**Comment by notmart** [comment]
hmm, so those runners would be services of their own then? (or a pretty brutal way would be.. just execute a command and take output? but would be very ugly)

**Comment by star-buck** [comment]
@aleixpol : sounds interesting, any pro and cons? 
I assume it doesnt change krunner backwards-compatibility?
How much work would that be?

**Comment by aleixpol** [comment]
I'm a bit concerned that for every key stroke on KRunner we would end up with 20+ queries to store.kde.org (one for each KNS resource...).

**Comment by bhush9** [comment]
no? AFAIU, this is about having way to distribute "runners" over store.kde.org and not searching resources on store.kde.org..?

**Comment by notmart** [comment]
yes, it's about having runners in a scripted only language since would be the only way to have them installed by gethotnewstuff. python bindings seems to really not be a thing anymore and writing javascript bindings for krunner wouldn't permit to write particularly interesting ones.

so this idea i guess would be to have a "meta" runner that starts external processes (tipically written in python i guess) and builds results over that i don't know how safe would be or how feasible...
nice thing would be that the external process can crash as much as it wants, i guess would be fairly easy to build results out of it, but fairly hard then to "run" them as needs a state preservation between different runs of this process

**Comment by aleixpol** [comment]
Ugh, yes, sorry.

> but fairly hard then to "run" them as needs a state preservation between different runs of this process
Maybe we can make it part of the protocol to specify the action as a URI and then trigger it by sending the URI back to the application.

Also I wouldn't tie it to any technology really, people like using random languages at random moments of their life, so documenting a dbus (or otherwise) protocol sounds safest to me.

Moving it into a separate process also allows the runners to be sandboxes easily.

**Comment by notmart** [comment]
hideally i would even like to have a krunnerd or whatever in one process and have the clients use dbus exclusively... I think IVan was working on something like that before being reasigned to per-activity favorites, once that is merged and done maybe that work could be revisited
